Linear-and-logarithmic combined scale for y-scales.

inkdot7 opened the pull request:
This merge request is on top of #3 .  Please first consider that.

Note that the `nonlinear` option requires `gnuplot 5.2`.

This patch very likely needs some further tuning and test before being reasonable for use.

inkdot7 committed:
Missing files for linear-and-logarithmic combined scale for y-scales.

ddrown reviewed (changes requested):
This looks interesting, but linscalelimit and linlogytics are missing

inkdot7 committed:
Linear-and-logarithmic combined scale for y-scales.

Requires gnuplot 5.2.  (nonlinear)

Many plots are most interesting for small values around 0, but the scale
may be given by some few outliers.  Some remote source may be good and
thus have values close to zero, while other have much larger values.  Then
the larger determine the plot scales, making the details of the good
remote sources invisible.

Due to the values taking both positive and negative values, and also at
or close to zero, the use of a pure logarithmic scale is not possible.

Instead, an inner linear range is chosen, and then both above and below
that logarithmic scales are used.

The range of the linear scale is chosen thus: for each plotted curve
determine the 50th percentile absolute value.  This ignores
the worst outliers of the curve.  The curve of a plot with the smallest
such percentile is chosen.  The linear range is then set as the smallest
value of 10^n * (1, 2 or 5) that is just larger than the limit.

The y tic marks are forced such that 0, 0.5*limit and limit are shown,
as well as limit*10^(1, 2, 3, ... 6).  for both positive and negative
values.

No ranges are set, the gnuplot automatic range selection works, and e.g.
if only a large logarithmic part is needed on the positive or negative
side, only that is shown.

inkdot7 commented:
Sorry about the missing files...
I've added them.  Also rebased for the changes just now merged to master.
However, have not tested - was some while ago I used chrony-graph last time :)
Perhaps you can have a look and see how bad it is?  :)

ddrown commented:
Works for me! Thanks!
